src/lib/storage/partitioning/partition.cpp

```diff
@@ -0,0 +1,26 @@
+#include "storage/partitioning/partition.hpp"
+#include "resolve_type.hpp"
+#include "storage/reference_column.hpp"
+
+namespace opossum {
+
+Partition::Partition(PartitionID partition_id) : _partition_id(partition_id) {}
+
+const PartitionID Partition::get_partition_id() { return _partition_id; }
+
+void Partition::clear() { _chunks.clear(); }
+
+void Partition::add_new_chunk(std::shared_ptr<Chunk> chunk) { _chunks.emplace_back(chunk); }
+
+void Partition::append(const std::vector<AllTypeVariant>& values) { _chunks.back()->append(values); }
+
+std::vector<std::shared_ptr<const Chunk>> Partition::get_chunks() const {
```

Comment:
Very considerate of you to create a list of immutable Chunks here. But we do not do this anywhere else, so its probably best, for consistency's sake, to just return the _chunks vector here.